### PR TITLE
Fixed issue where nested responsive mixins were outputting invalid MQs

### PR DESCRIPTION
**Pull request by jamesjacobs**
**Background**
As noticed in #697 forms were not 100% width on tablet widths in IE. On investigation, this was down to how sass compiles nested mixins where both contain media queries.

For example
```
.tab__container.has-sidebar .tab__content {
    @include respond-min($screen-phone) {
        @include col-span(12);
    }

    @include respond-min($screen-desktop) {
        @include col-span(9);
    }
}
```
Would result in invalid MQ when compiled. The above would compile to:

```
@media screen and (min-width: 480px) and (min-width: 480px) {
  .tab__container.has-sidebar .tab__content {
    width: 100%; } }

@media screen and (min-width: 480px) and (min-width: 767px) {
  .tab__container.has-sidebar .tab__content {
    width: 100%; } }

@media screen and (min-width: 480px) and (min-width: 992px) {
  .tab__container.has-sidebar .tab__content {
    width: 100%; } }

@media screen and (min-width: 992px) and (min-width: 480px) {
  .tab__container.has-sidebar .tab__content {
    width: 100%; } }

@media screen and (min-width: 992px) and (min-width: 767px) {
  .tab__container.has-sidebar .tab__content {
    width: 74.875%; } }

@media screen and (min-width: 992px) and (min-width: 992px) {
  .tab__container.has-sidebar .tab__content {
    width: 74.875%; } }
```
Note the duplicate `min-width`'s.

**Fix**
There was a couple of options to fix. One of which was to remove all MQs from the `col-span` mixin. However, due to needing to support the `force` and `gutter` options this wasn't possible. Instead, I created a new `col-span-width` mixin which only set the width based on grid columns and doesn't contain any MQs.

This resulted in `719` lines of unnecessary and duplicate MQ CSS been removed from the compiled CSS file - prefix 17083 lines, postfix 16364 lines 🎉

Whilst fixing this I noticed a couple of other MQ grid-related issues which have been fixed detailed below:

Orginal issue:
33260521-fbc35fa4-d357-11e7-975e-6da8a29af3f7

Fix:
ie form width fix

Select2 mobile issue (left) and fix (right):
select2 before and after fix

Grid col mobile issue:
grid on mobile

Grid col mobile fix:
grid mobile fix

Choice block mobile issue:
choice margins on mobile issue

Choice block mobile fix:
choice margins on mobile fix

Browser tested in:

- [x] Chrome 
- [x] Firefox 
- [x] Safari
- [x] Edge 
- [x] IE11 
- [x] IE10 
- [x] IE9

### CODE REVIEW

File: stylesheets/_component.masterswitch.scss [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+		display: table-cell;
+
+		@include respond-min($screen-tablet) {
+			@include col-span-width($form-label-cols);
+			display: inline-block;
 		}
 	}
 }
 
+
```
Comment (nvasileiadis) [inline stylesheets/_component.masterswitch.scss]: Unnecessary extra line?

File: stylesheets/_structure.grid.scss [hunk 2]
```diff
@@ -62,14 +62,25 @@ $gutter_pc:     .5;
...
-            width: $span_width;
-        }
-        @else {
-            width: $span_width !important;
-        }
+/**
+ * Assigns width based on number of grid columns. Does not provide responsive styling, use the col-span mixin for
+ * built in responsiveness. Useful when assigning grid width to an element within a media query.
+ *
+ * @param  {int}  $num                   The number of columns the width should span
```
Comment (nvasileiadis) [inline stylesheets/_structure.grid.scss]: Extra space before $num

File: stylesheets/_structure.grid.scss [hunk 3]
```diff
@@ -62,14 +62,25 @@ $gutter_pc:     .5;
...
+ * @param  {int} $columns: $columns      The total number of grid columns to use for grid calculations, defaults to $columns (12)
+ * @param  {bool} $important:  false     If true, adds the !important declaration
+ */
+@mixin col-span-width($num, $columns: $columns, $important: false) {
+    $one_col: (100% - ($gutter_pc * ($columns - 1))) / $columns;
+    $span_width: ($one_col * ($num)) + ($gutter_pc * ($num - 1));
+
+    @if $important == false {
```
Comment (nvasileiadis) [inline stylesheets/_structure.grid.scss]: Maybe '===' instead of '=='

File: stylesheets/_component.masterswitch.scss (hunk 1 above)
Comment (jamesjacobs) [reply stylesheets/_component.masterswitch.scss]: Removed

File: stylesheets/_structure.grid.scss (hunk 2 above)
Comment (jamesjacobs) [reply stylesheets/_structure.grid.scss]: Used for alignment

File: stylesheets/_structure.grid.scss (hunk 3 above)
Comment (jamesjacobs) [reply stylesheets/_structure.grid.scss]: There is no`===` in sass.

**Comment by jamesjacobs** [comment]
closes #697

**Pull request merged**